rasterio/plot.py

```diff
@@ -105,31 +105,34 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
                 # will have the colors in the order expected by
                 # matplotlib (RGB)
                 arr = source.read(rgb_indexes, masked=True)
-                arr = reshape_as_image(arr)
-
             except KeyError:
-                arr = source.read(1, masked=True)
+                if indexes == None: indexes = (0,1,2)
+                indexes = tuple(i+1 for i in indexes)
+                arr = source.read(indexes, masked=True)
+
+            arr = reshape_as_image(arr)
     else:
         # The source is a numpy array reshape it to image if it has 3+ bands
         if source.ndim >= 3:
             source = np.ma.squeeze(source)
 
         if source.ndim >= 3:
             arr = reshape_as_image(source)
+            if indexes == None: indexes = (0,1,2)
+            if arr.shape[-1]>3: arr = arr[:, :, indexes]
+
         else:
             arr = source
 
         if transform and with_bounds:
             kwargs['extent'] = plotting_extent(arr, transform)
 
-    if adjust and arr.ndim >= 3:
-        # Adjust each band by the min/max so it will plot as RGB.
-        arr = reshape_as_raster(arr).astype("float64")
-
-        for ii, band in enumerate(arr):
-            arr[ii] = adjust_band(band)
-
-        arr = reshape_as_image(arr)
+    if strech:
+        if clip_percent: 
+            clip_percent=clip_percent
+        else:
+            clip_percent=2
+        arr = hist_strech(arr, clip_percent)
```

Comment:
```suggestion
        clip_percent = clip_percent or 2
        arr = hist_strech(arr, clip_percent)
```

Reply:
the scripts have been modified according to your suggestion.